packages/core/src/config.ts

```diff
@@ -157,6 +158,7 @@ export function resolveConfig<Theme extends object = object>(
     extractors,
     safelist: getMerged('safelist'),
     separators,
+    details: config.details ?? process.env.NODE_ENV === 'development',
```

Comment:
The core is env agnostic, we shouldn't rely on process. Might be better to rely on envMode and move this logics to the vite/webpack plugin